xbmc/utils/IBufferObject.h

```diff
@@ -15,12 +15,13 @@ class IBufferObject
 public:
   virtual ~IBufferObject() = default;
 
-  virtual bool CreateBufferObject(int width, int height) = 0;
-  virtual void DestroyBufferObject() { }
+  virtual bool CreateBufferObject(int format, int width, int height) = 0;
```

Comment:
This comment actually applies to this line added in https://github.com/xbmc/xbmc/pull/14079. I see you've used int for dimensions, but I don't see characters `< 0` anywhere. Red flag. These dimensions can come from outside `xbmc/utils/`, so we need to guard against bad input due to failures in remote parts of the code. I propose you change these to unsigned, which will strengthen the assumptions presented to outside code.

Reply:
Updated, totally makes sense. I've also changed width and height to unsigned.